### PR TITLE
Rework types (option 3)

### PR DESCRIPTION
**Pull request by dmlloyd**
* Create constants for each type ID
* Use switch statements for finding types by ID to avoid boxing overheads
* Add query methods for type characteristics
* Replace `ElementType` with `ValueType`

### CODE REVIEW

File: wasm/src/main/java/com/dylibso/chicory/wasm/types/ValueType.java
Comment (andreaTP) [inline wasm/src/main/java/com/dylibso/chicory/wasm/types/ValueType.java:147]: I think that this can be the only default implementation, I would call it `forId` and make the current `forId` `private` or simply remove it, wdyt?

File: wasm/src/main/java/com/dylibso/chicory/wasm/types/ValueType.java
Comment (andreaTP) [inline wasm/src/main/java/com/dylibso/chicory/wasm/types/ValueType.java:174]: What about keeping only this as the default implementation for `refTypeForId` ?

**Review by andreaTP — CHANGES_REQUESTED** [review]
I like the changes done in this direction, I would remove the `OrThrow` suffix and make the throwing methods the only and default implementation.
Since we are consolidating the API I think we can go down this path 🙂

**Comment by dmlloyd** [comment]
Applied suggestions to not have a `null`-returning option for the get-by-ID methods.

**Review by andreaTP — APPROVED** [review]
This looks good to me, but I would like to have a second opinion before merging if it doesn't take too long.

cc. @bhelx @danielperano @mariofusco

**Comment by dmlloyd** [comment]
Looks like there's one problem due to the non-null returning option, but the fix should be simple.

**Comment by andreaTP** [comment]
(and CI should be green 😄 )

**Comment by dmlloyd** [comment]
All fixed :+1:

**Review by mariofusco — APPROVED** [review]

**Pull request merged**
